### ISSUE TITLE
Caught panic with message: Unknown error type, can't produce message

### ISSUE THREAD

**Issue by doublemcz** [Spotifyd/spotifyd#125]
After couple of hours I wanted to connect and play (I selected the device on iPhone in the Spotify App). I hit play, nothing happened. I tried it again, after 15 seconds and third try it started to work. The /var/log/syslog shows error `Caught panic with message: Unknown error type, can't produce message`

Platform: Raspberry PI (downloaded compiled image)
Running as background system daemon.

Whole log around the error:

```
pr 12 18:41:30 localhost spotifyd[15446]: 18:41:30 [ERROR] Caught panic with message: Unknown error type, can't produce message.
Apr 12 18:41:30 localhost systemd[1]: spotifyd.service: Main process exited, code=exited, status=101/n/a
Apr 12 18:41:30 localhost systemd[1]: spotifyd.service: Unit entered failed state.
Apr 12 18:41:30 localhost systemd[1]: spotifyd.service: Failed with result 'exit-code'.
Apr 12 18:41:42 localhost systemd[1]: Created slice User Slice of martin.
Apr 12 18:41:42 localhost systemd[1]: Started Session c4 of user martin.
Apr 12 18:41:42 localhost systemd[1]: Starting User Manager for UID 1001...
Apr 12 18:41:42 localhost systemd[1]: spotifyd.service: Service hold-off time over, scheduling restart.
Apr 12 18:41:42 localhost systemd[1]: Stopped A spotify playing daemon.
Apr 12 18:41:42 localhost systemd[1]: Started A spotify playing daemon.
```

**Comment by benjamincr** [comment]
Same issue here, but for some reason my systemd service doesn't stop and tells me the daemon is still active, which means it doesn't re start automatically and I have to ssh to my music server to do it.

This is quite annoying for a daemon that otherwise works exactly like it should.

**Comment by zjttoefs** [comment]
I get the same message after some minutes of playing now. This is with debugging:
```
May 13 20:06:04 musicpi2 spotifyd[8204]: 20:06:04 [DEBUG] tokio_core::reactor: loop process, Duration { secs: 0, nanos: 83385 }
May 13 20:06:27 musicpi2 spotifyd[8204]: 20:06:27 [TRACE] tokio_reactor: [<unknown>:330] event Readable | Writable | Error | Hup Token(6)
May 13 20:06:27 musicpi2 spotifyd[8204]: 20:06:27 [DEBUG] tokio_reactor: loop process - 1 events, 0.000s
May 13 20:06:27 musicpi2 spotifyd[8204]: 20:06:27 [DEBUG] librespot_core::session: drop Dispatch
May 13 20:06:27 musicpi2 spotifyd[8204]: 20:06:27 [TRACE] mio::poll: [<unknown>:905] deregistering handle with poller
May 13 20:06:27 musicpi2 spotifyd[8204]: 20:06:27 [DEBUG] tokio_reactor: dropping I/O source: 5
May 13 20:06:27 musicpi2 spotifyd[8204]: 20:06:27 [ERROR] Caught panic with message: Unknown error type, can't produce message.
May 13 20:06:27 musicpi2 spotifyd[8204]: 20:06:27 [DEBUG] librespot_connect::spirc: drop Spirc[0]
May 13 20:06:27 musicpi2 spotifyd[8204]: 20:06:27 [DEBUG] librespot_playback::player: Shutting down player thread ...

```

**Comment by praul** [comment]
Same issue with me. If internet disconnects, spotifyd stops working. Also, systemd does not restart the service. Im helping myself with a little scripts that polls journalctl for errors, but a solid fix would be greatly appreciated. spotifyd running and restarting, no matter what - that would be perfect

**Comment by slondr** [comment]
Should be resolved by #511 which was included in today's v0.2.25 release.

If this problem pops up again, let us know.
